### PR TITLE
fix #104

### LINKED ISSUE #104: change type of zn_cursor::x,y from int to double

### PR DESCRIPTION
**Pull request by watasuke102**
## Context

See #104 

## Summary

Change type: cursor position int -> double, screen box wlr_box -> wlr_fbox

### CODE REVIEW

File: zen/scene/screen-layout.c [hunk 1]
```diff
@@ -6,25 +6,56 @@
...
+  // find closest y point
+  if (y < box->y) {
+    *dest_y = box->y;
+  } else if (y >= box->y + box->height) {
+    *dest_y = box->y + box->height - 1;
+  } else {
+    *dest_y = y;
+  }
+}
+
```
Comment (Aki-7) [inline zen/scene/screen-layout.c:39]: Adding function starting with `wlr` is confusing.

How about creating `zn_wlr_fbox_closest_point` in `zen/wlr/box.c` and `include/zen/wlr/box.h`?

File: zen/scene/screen-layout.c [hunk 2]
```diff
@@ -6,25 +6,56 @@
...
+    *dest_x = NAN;
+    *dest_y = NAN;
+    return;
+  }
+
+  // find the closest x point
+  if (x < box->x) {
+    *dest_x = box->x;
+  } else if (x >= box->x + box->width) {
+    *dest_x = box->x + box->width - 1;
```
Comment (Aki-7) [inline zen/scene/screen-layout.c:25]:
```suggestion
    *dest_x = box->x + box->width;
```

File: zen/scene/screen-layout.c [hunk 3]
```diff
@@ -6,25 +6,56 @@
...
+  } else {
+    *dest_x = x;
+  }
+
+  // find closest y point
+  if (y < box->y) {
+    *dest_y = box->y;
+  } else if (y >= box->y + box->height) {
+    *dest_y = box->y + box->height - 1;
```
Comment (Aki-7) [inline zen/scene/screen-layout.c:34]:
```suggestion
    *dest_y = box->y + box->height;
```

**Review by Aki-7 — CHANGES_REQUESTED** [review]

File: zen/scene/screen-layout.c (hunk 1 above)
Comment (watasuke102) [reply zen/scene/screen-layout.c]: Nice suggestion, and I also added `zn_wlr_fbox_contains_point()`.

See: https://github.com/zigen-project/zen/pull/105#discussion_r946338563

File: include/zen/wlr/box.h [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+#define ZEN_BOX_H
+
+#include <wlr/util/box.h>
+
+bool zn_wlr_fbox_contains_point(const struct wlr_fbox* box, double x, double y);
+
+void zn_wlr_fbox_closest_point(const struct wlr_fbox* box, double x, double y,
+    double* dest_x, double* dest_y);
+
+#endif  // ZEN_BOX_H
```
Comment (Aki-7) [inline include/zen/wlr/box.h:11]:
```suggestion
#endif  // ZEN_BOX_H

```

another nits

**Review by Aki-7 — CHANGES_REQUESTED** [review]

**Review by Aki-7 — APPROVED** [review]
LGTM!!

**Pull request merged**